pkg/sdk/external_functions_gen_test.go

```diff
@@ -0,0 +1,250 @@
+		opts.Compression = String("GZIP")
+		rt := NewSchemaObjectIdentifier("db", "schema", "request_translator")
+		opts.RequestTranslator = &rt
+		rs := NewSchemaObjectIdentifier("db", "schema", "response_translator")
+		opts.ResponseTranslator = &rs
+		opts.As = "https://xyz.execute-api.us-west-2.amazonaws.com/prod/remote_echo"
+		assertOptsValidAndSQLEquals(t, opts, `CREATE OR REPLACE SECURE EXTERNAL FUNCTION %s (id NUMBER, name VARCHAR) RETURNS VARCHAR NOT NULL CALLED ON NULL INPUT IMMUTABLE COMMENT = 'comment' API_INTEGRATION = "api_integration" HEADERS = ('header1' = 'value1', 'header2' = 'value2') CONTEXT_HEADERS = (CURRENT_ACCOUNT, CURRENT_USER) MAX_BATCH_ROWS = 100 COMPRESSION = GZIP REQUEST_TRANSLATOR = %s RESPONSE_TRANSLATOR = %s AS 'https://xyz.execute-api.us-west-2.amazonaws.com/prod/remote_echo'`, id.FullyQualifiedName(), rt.FullyQualifiedName(), rs.FullyQualifiedName())
+	})
+}
+
+func TestExternalFunctions_Alter(t *testing.T) {
+	id := RandomSchemaObjectIdentifier()
+
+	defaultOpts := func() *AlterExternalFunctionOptions {
+		return &AlterExternalFunctionOptions{
+			name:              id,
+			IfExists:          Bool(true),
+			ArgumentDataTypes: []DataType{DataTypeVARCHAR, DataTypeNumber},
+		}
+	}
+
+	t.Run("validation: nil options", func(t *testing.T) {
+		opts := (*AlterExternalFunctionOptions)(nil)
+		assertOptsInvalidJoinedErrors(t, opts, ErrNilOptions)
+	})
+
+	t.Run("validation: incorrect identifier", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.name = NewSchemaObjectIdentifier("", "", "")
+		assertOptsInvalidJoinedErrors(t, opts, ErrInvalidObjectIdentifier)
+	})
+
+	t.Run("validation: exactly one field should be present", func(t *testing.T) {
+		opts := defaultOpts()
+		assertOptsInvalidJoinedErrors(t, opts, errExactlyOneOf("AlterExternalFunctionOptions", "Set", "Unset"))
+	})
+
+	t.Run("validation: exactly one field should be present", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.Set = &ExternalFunctionSet{
+			MaxBatchRows: Int(100),
+		}
+		opts.Unset = &ExternalFunctionUnset{
+			MaxBatchRows: Bool(true),
+		}
+		assertOptsInvalidJoinedErrors(t, opts, errExactlyOneOf("AlterExternalFunctionOptions", "Set", "Unset"))
+	})
+
+	t.Run("alter: set", func(t *testing.T) {
+		opts := defaultOpts()
+		integration := NewAccountObjectIdentifier("api_integration")
+		opts.Set = &ExternalFunctionSet{
+			ApiIntegration: &integration,
+		}
+		assertOptsValidAndSQLEquals(t, opts, `ALTER FUNCTION IF EXISTS %s (VARCHAR, NUMBER) SET API_INTEGRATION = "api_integration"`, id.FullyQualifiedName())
+
+		opts = defaultOpts()
+		opts.Set = &ExternalFunctionSet{
+			Headers: []ExternalFunctionHeader{
+				{
+					Name:  "header1",
+					Value: "value1",
+				},
+				{
+					Name:  "header2",
+					Value: "value2",
+				},
+			},
+		}
+		assertOptsValidAndSQLEquals(t, opts, `ALTER FUNCTION IF EXISTS %s (VARCHAR, NUMBER) SET HEADERS = ('header1' = 'value1', 'header2' = 'value2')`, id.FullyQualifiedName())
```

Comment:
I think I prefer tests split rather than put down into one function (I mean 1 test == setup + asserts, instead of alternating setup + assets).

Reply:
fixed